src/ChangeStream.php

```diff
@@ -82,7 +87,7 @@ public function rewind()
 
     public function valid()
     {
-
+        return $this->csIt->key() !== null;
```

Comment:
Looking at [`IteratorIterator::key()`](https://github.com/php/php-src/blob/PHP-7.1.11/ext/spl/spl_iterators.c#L1774), I believe this is equivalent to returning [`IteratorIterator::valid()`](https://github.com/php/php-src/blob/PHP-7.1.11/ext/spl/spl_iterators.c#L1753). If you think that's more readable, you can swap to returning `$this->csIt->valid()`, but this should work perfectly fine as-is.

Both of them depend on [`spl_dual_it_fetch()`](https://github.com/php/php-src/blob/PHP-7.1.11/ext/spl/spl_iterators.c#L1693) to populate the current data and key from the inner iterator (our Cursor). Looking at [Cursor's key handler](https://github.com/mongodb/mongo-php-driver/blob/1.3.2/src/MongoDB/Cursor.c#L69) again, I realize I misspoke earlier when I implied that it could return `null` if there was no current element. It actually always returns an integer. That's not technically a problem since we don't implement the userland Iterator interface. PHP's IteratorIterator class and its `key()` and `valid()` methods still function properly because our [current_data handler](https://github.com/mongodb/mongo-php-driver/blob/1.3.2/src/MongoDB/Cursor.c#L77) _does_ return an undefined value if there is no current element. This is because we always free/unset the current element before advancing with the next handler. If libmongoc's cursor doesn't have a next element for us, the Cursor's current element is left unset.